internal/pkg/manifest/validate.go

```diff
@@ -351,18 +352,51 @@ func (a *AdvancedCount) Validate() error {
 	if a.Spot != nil && a.hasAutoscaling() {
 		return &errFieldMutualExclusive{
 			firstField:  "spot",
-			secondField: "range/cpu_percentage/memory_percentage/requests/response_time",
+			secondField: "range/cpu_percentage/memory_percentage/requests/response_time/queue_delay",
 		}
 	}
 	if err := a.Range.Validate(); err != nil {
 		return fmt.Errorf(`validate "range": %w`, err)
 	}
-	if a.Range.IsEmpty() && (a.CPU != nil || a.Memory != nil || a.Requests != nil || a.ResponseTime != nil) {
+	if a.Range.IsEmpty() && (a.CPU != nil || a.Memory != nil || a.Requests != nil || a.ResponseTime != nil || !a.QueueScaling.IsEmpty()) {
 		return &errFieldMustBeSpecified{
 			missingField:      "range",
-			conditionalFields: []string{"cpu_percentage", "memory_percentage", "requests", "response_time"},
+			conditionalFields: []string{"cpu_percentage", "memory_percentage", "requests", "response_time", "queue_delay"},
 		}
 	}
+	if err := a.QueueScaling.Validate(); err != nil {
+		return fmt.Errorf(`validate "queue_delay": %w`, err)
+	}
+	return nil
+}
+
+// Validate returns nil if QueueScaling is configured correctly.
+func (qs *QueueScaling) Validate() error {
+	if qs.IsEmpty() {
```

Comment:
I wonder if it's better to call `Validate()` only when `!struct.IsEmpty()`, or consider `IsEmpty()` as valid and call `Validate()` whether the struct is empty or not.

Reply:
I opted this route because I didn't want the client to think about whether it's safe to call `Validate` or not.
This way client code becomes easy to reason about:
```
go through each field:
  call Validate()
```

Reply:
Actually I kind of get what wanxian said, if we return `nil` when the struct is empty, we cannot safely use the struct after validation because it could be an empty one. No matter we call `isEmpty` or not in `Validate()`, the caller needs to make sure the struct is not empty.

Overall I don't think we should use `Validate()` out of the `validate.go` file. Any thought?

Reply:
I'll change it to validate only if the queue is not empty, but I think it should be okay for anybody to call the `Validate` function like in `AcceptableBacklog`.